### PR TITLE
Create search bar component

### LINKED ISSUE #197: [Projects screen] Crate a search bar component
Use debounce

### PR DESCRIPTION
**Pull request by Noamili**
close #197
TODO Change target branch to staging 

Default value:
Screen Shot 2022-07-03 at 20 47 12

When searching:
Screen Shot 2022-07-03 at 20 47 26
@MichalPorag updated design

### CODE REVIEW

File: components/searchBar/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const debounce = (callback, delay) => {
+  let timer;
+  return (...args) => {
+    clearTimeout(timer);
+    timer = setTimeout(() => callback(...args), delay);
+  };
+};
+
+const SearchBar = (props: Props): ReactElement => {
```
Comment (MichalPorag) [inline components/searchBar/SearchBar.tsx:16]: Please add an option to pass placeholder text

File: components/searchBar/SearchBar.module.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  svg {
+    margin: 10px 16px;
+  }
+}
+
+.textBox {
+  border: none;
+  width: 100%;
+  border-radius: 8px;
+  padding-right: 10px;
```
Comment (MichalPorag) [inline components/searchBar/SearchBar.module.scss:18]: Please use a gap in the container instead.

File: components/searchBar/SearchBar.module.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  margin: 100px;
+  svg {
+    margin: 10px 16px;
+  }
+}
+
+.textBox {
+  border: none;
+  width: 100%;
+  border-radius: 8px;
```
Comment (MichalPorag) [inline components/searchBar/SearchBar.module.scss:17]: Why does the text nee border-radius?

File: components/searchBar/SearchBar.module.scss [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  padding-right: 10px;
+  direction: ltr;
```
Comment (MichalPorag) [inline components/searchBar/SearchBar.module.scss:19]: Isn't it the default?

File: components/searchBar/SearchBar.module.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
+.textBoxContainer {
+  display: flex;
+  border: solid 1px #bfc0c3;
+  max-width: 313px;
+  height: 40px;
+  border-radius: 8px;
+  background-color: white;
+  margin: 100px;
+  svg {
```
Comment (MichalPorag) [inline components/searchBar/SearchBar.module.scss:9]: Please moe the padding logic from SVG to the textBoxContainer class

**Review by MichalPorag — CHANGES_REQUESTED** [review]
Great work, just a few comments :)

File: components/searchBar/SearchBar.module.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
+.textBoxContainer {
```
Comment (Noamili) [inline components/searchBar/SearchBar.module.scss:1]: 16 px padding on sides and gap of 8px, center

File: components/searchBar/SearchBar.module.scss (hunk 3 above)
Comment (Noamili) [reply components/searchBar/SearchBar.module.scss:17]: in the right side, the input element spread until the end of the container div, so its corners needed to be rounded.
now we have padding on the sides of the container so this line can be removed

File: components/searchBar/SearchBar.module.scss (hunk 4 above)
Comment (Noamili) [reply components/searchBar/SearchBar.module.scss:19]: in styles/globals.scss the direction is rtl

**Review by MichalPorag — APPROVED** [review]

**Pull request merged**